AK/String.h

```diff
@@ -259,3 +259,21 @@ struct Formatter<String> : Formatter<StringView> {
 };
 
 }
+
+[[nodiscard]] ALWAYS_INLINE AK::ErrorOr<AK::String> operator""_string(char const* cstring, size_t length)
+{
+    return String::from_utf8(AK::StringView(cstring, length));
+}
+
+// FIXME: Remove this when clang fully supports consteval (specifically in the context of default parameter initialization).
+// See: https://stackoverflow.com/questions/68789984/immediate-function-as-default-function-argument-initializer-in-clang
+#if defined(AK_COMPILER_CLANG)
+#    define AK_SHORT_STRING_LITERAL_CONSTEVAL constexpr
+#else
+#    define AK_SHORT_STRING_LITERAL_CONSTEVAL consteval
+#endif
+
+[[nodiscard]] ALWAYS_INLINE AK_SHORT_STRING_LITERAL_CONSTEVAL AK::String operator""_short_string(char const* cstring, size_t length)
+{
+    return String::from_utf8_short_string(AK::StringView(cstring, length));
```

Comment:
Can we static-assert here that length is short enough to fit?

Reply:
The `String::from_utf8_short_string()` function itself asserts that.

Reply:
`from_utf8_short_string` raises a compiler error itself if it's too long

Reply:
Oh cool. :+1:

Reply:
We cannot:

```
/home/linus/Dev/serenity/Meta/Lagom/../../AK/String.h: In function ‘consteval AK::String operator""_short_string(const char*, size_t)’:
/home/linus/Dev/serenity/Meta/Lagom/../../AK/String.h:278:26: error: non-constant condition for static assertion
  278 |     static_assert(length <= String::MAX_SHORT_STRING_BYTE_COUNT);
      |                   ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/linus/Dev/serenity/Meta/Lagom/../../AK/String.h:278:26: error: ‘length’ is not a constant expression
```

Reply:
Ah, right, I guess because it's `consteval` the upstream `VERIFY()` does the trick!